test/tools/evmoneRunner.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/evmoneRunner.h>
+#include <iostream>
+#include <assert.h>
+#include <fstream>
+#include <limits>
+
+using namespace external::evmone;
+using namespace std;
+
+evmc_host_interface EvmOneVM::interface{
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		[](evmc_context* ctx) { return static_cast<EvmOneVM*>(ctx)->tx_context; },
```

Comment:
```suggestion
		[](evmc_context* ctx) { return dynamic_cast<EvmOneVM&>(*ctx).tx_context; },
```

Reply:
The thing is that `evmc_context` is a non-polymorphic struct, so dynamic cast is not possible.

```
error: 'evmc_context' is not polymorphic
                [](evmc_context* ctx) { return dynamic_cast<EvmOneVM&>(*ctx).m_txContext; },
                                               ^                       ~~~~
```